### DISCUSSION TITLE
Disable telemetry with `ASTRO_TELEMETRY_DISABLED=1` environment variable

### DISCUSSION THREAD

**Discussion by jlarmstrongiv** [withastro/roadmap#405] [Proposal]
Many frameworks support disabling telemetry with environment variables:

```bash
# nextjs
export NEXT_TELEMETRY_DISABLED=1
# prisma
export CHECKPOINT_DISABLE=1
# serverless
export SLS_TELEMETRY_DISABLED=1
# storybook
export STORYBOOK_DISABLE_TELEMETRY=1
# aws sam
export SAM_CLI_TELEMETRY=0
# homebrew
export HOMEBREW_NO_ANALYTICS=1
# vscode vue
export VUEDX_TELEMETRY='off'
```

Using environment variables makes it easy to disable analytics with dotfiles for personal work on multiple machines or professional work with secrets managers for multiple developers.

Currently, astro supports disabling telemetry with [`astro telemetry disable
`](https://docs.astro.build/en/reference/cli-reference/#astro-telemetry), but using environment variables would ensure opt-out for every environment, machine, and developer more easily.

**Comment by Princesseuh** [comment]
Hey, this is actually supported! You can use either `ASTRO_TELEMETRY_DISABLED` or `TELEMETRY_DISABLED`. It seems like this is not documented however. cc @delucis

**Reply by delucis** [reply]
Hah! I had a vague feeling it was but then didn't find it in the docs. 😅

**Reply by kireerik** [reply]
How can I verify (make sure) that when I set it then it is actually disabled?

Does the following works in the Astro Configuration File?:
```Node.js
defineConfig({
	vite: {
		define: {
			TELEMETRY_DISABLED: true
		}
	}
})
```

**Reply by delucis** [reply]
My guess would be this needs to be set _before_ Astro starts up, so defining via the Astro config wouldn’t work, but will double check. That would mean setting it on the CLI, e.g.
```sh
ASTRO_TELEMETRY_DISABLED=true npm run build
```
Or by setting the variable in a CI environment dashboard for those situations.

**Reply by kireerik** [reply]
When I set `TELEMETRY_DISABLED` in `.env` I can see that on `import.meta.env`. However, if I set anything else (in `.env`) (including `ASTRO_TELEMETRY_DISABLED`) I do not see them (on `import.meta.env`).

So I am not sure which one should I use and whether they have the desired effect at all.

**Comment by kireerik** [comment]
Should setting either of the following (for `defineConfig`) work?:
- `forceDisableTelemetry` (to `true`)
- `disableTelemetry` (to `true`)
- `telemetry` (to `false`)
